src/coreclr/jit/rationalize.cpp

```diff
@@ -99,6 +99,23 @@ void Rationalizer::RewriteNodeAsCall(GenTree**             use,
             // for intrinsics that get rewritten back to user calls
             assert(!operand->OperIsFieldList());
 
+            if (varTypeIsMask(operand->TypeGet()))
+            {
+#if defined(FEATURE_HW_INTRINSICS)
```

Comment:
Instead wrap the if-condition in `#ifdef FEATURE_MASKED_HW_INTRINSICS`

Reply:
`TYP_MASK` and a lot of the surrounding support isn't under that feature switch.

If we were to centralize everything under that feature switch, I think it should be a separate PR. However, I'm also not convinced we need that as we didn't set any such flag up for XARCH.

Reply:
I don't believe we do today, but that doesn't mean it will never happen. `varTypeIsMask` returns constant `false` on unsupported platforms, so this just ensures any future scenarios handle this appropriately and will be optimized out as dead code otherwise.